main_test.go

```diff
@@ -56,11 +52,16 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		log.Fatalf("failed to setup Postgres container")
 	}
+
 	customerRepository, err = NewCustomerRepository(ctx, pgContainer.ConnectionString)
 	if err != nil {
 		log.Fatalf("failed to initialize customerRepository")
 	}
-	defer pgContainer.CloseFn()
+	defer func() {
+		if err := pgContainer.Terminate(ctx); err != nil {
+			log.Fatalf("error terminating postgres container: %s", err)
```

Comment:
Seeing the log.Fatalf execution here in the context of the test initialisation, makes it clearer that that test will stop in the case the termination does not happen

Reply:
When using testify suite, we want to call this from `TearDownSuite()` where we don't have access to context. Should we use `context.Background()`?

Reply:
I'd probably have a reference to the context in the suite, so that we could eventually pass it from test to test.